pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,173 @@
+import { EmptyState, EmptyStateBody } from '@patternfly/react-core/dist/esm/components/EmptyState/index.js';
+import { Grid } from '@patternfly/react-core/dist/esm/layouts/Grid/index.js';
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+import { MinusIcon } from '@patternfly/react-icons';
+
+import './wireguard.scss';
+
+const _ = cockpit.gettext;
+
+export function WireGuardDialog({ settings }) {
+    const idPrefix = 'network-wireguard-settings';
+    const model = useContext(ModelContext);
+    const Dialogs = useDialogs();
+    console.log(model);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [privateKey, setPrivateKey] = useState('');
+    const [publicKey, setPublicKey] = useState('');
+    const [listenPort, setListenPort] = useState('');
+    const [addresses, setAddresses] = useState('');
+    const [dialogError, setDialogError] = useState('');
+    const [peers, setPeers] = useState([]);
+
+    useEffect(() => {
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.script(`echo ${privateKey.trim()} | wg pubkey`, { err: 'message' });
+                setPublicKey(key.trim());
+            } catch (e) {
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [privateKey]);
+
+    function onSubmit() {
+        function createSettingsObj() {
+            return {
+                connection: {
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers,
+                },
+                ipv4: {
+                    addresses: [[addresses.split('/')[0], addresses.split('/')[1], '0.0.0.0']], // TODO: fix gateway
+                    method: 'manual'
+                }
+            };
+        }
+
+        dialogSave({
+            model,
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_('Private key')} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
+                    <Button variant='secondary'
+                        onClick={async () => {
+                            const key = await cockpit.script("wg genkey");
+                            setPrivateKey(key.trim());
```

Comment:
These 10 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230626-161008-14729dcc-fedora-38-devel-other//Coverage/lcov/github-pr.diff.gcov.html)